torch/_inductor/compile_fx.py

```diff
@@ -496,6 +496,15 @@ def fx_codegen_and_compile(
         )
         with V.set_graph_handler(graph):
             graph.run(*example_inputs)
+            context = torch._guards.TracingContext.get()
+            if context is not None and context.output_strides is not None:
+                # Return the output strides to the caller via TracingContext
+                assert len(context.output_strides) == 0
+                for out in graph.graph_outputs:
+                    if hasattr(out, "layout"):
```

Comment:
Do we see cases that `out` does not have a `layout` attribute? Does this happen for some SymInt returned?

Reply:
yes, can have symint return. Maybe there is something better than hasattr to do here, but a lot of inductor code is written this way. Hard to be more precise without typing 👹

Reply:
OK, turns out I can't, because strides in inductor are sympy.Symbol, not SymInt. So I am just going ahead and storing the hints here only. When you fix this to do permutations instead, probably can make this a little less fragile.

Reply:
We can not do permutations instead because of some non 'dense' activations? This happens for some real models.

Reply:
There are a few ways we can do this. If we only ever change layout on dense tensors, we can make `output_strides` be None *unless* we changed the stride. Then the permutation is always defined.

Reply:
> If we only ever change layout on dense tensors, ...

I think this can not be guaranteed in inductor right now.  A non dense tensor's layout may get changes because of the layout change of its upstream tensors.  And inductor does not force eager stride for that non dense tensor because of code here: https://github.com/pytorch/pytorch/blob/main/torch/_inductor/graph.py#L688 . I guess the algorithm will be tricky to restride a non-dense tensor.

Reply:
OK will need to think this through carefully. I'll start a doc.

Reply:
Upon further reflection, all of this is moot if we have compiled backwards, so let's not touch it unless it's causing someone problems.